---
title: "R Notebook"
output: html_notebook
editor_options: 
  chunk_output_type: console
---

---
title: "APAL_16S"
author: "matsuda"
date: "2023-03-01"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(warning=FALSE, message=FALSE)


library(plyr) 
library(dplyr)
library(ggplot2)
library(ggpubr)
library(lmerTest)
library(car)
library(emmeans)
library(gridExtra)
library(multcomp)
library(reshape)
library(factoextra)
library(reshape2)
library(vegan) 
library(pairwiseAdonis)
library("scales")
packageVersion("scales")
library(RColorBrewer)
library(colorRamps)
library(devtools)
library(phyloseq)
library(readr)
library(vegan)
library(ape)
library(geosphere)
library(ade4)
library(microbiome)  
library(knitr)
library(tidyverse)
```

#Load in data and get into phyloseq (save RData file)
```{r}       
#read in sample data
MetaData16<-read.csv("Apal_metadata.csv")
MetaData16$Group<-as.factor(MetaData16$Group)
MetaData16$Timepoint<-as.factor(MetaData16$Timepoint)
MetaData16$tileID<-as.factor(MetaData16$tileID)
MetaData16$tubeNumber<-as.factor(MetaData16$tubeNumber)
MetaData16$nursery<-as.factor(MetaData16$nursery)
MetaData16$parentID<-as.factor(MetaData16$parentID)
MetaData16$parent<-as.factor(MetaData16$parent)
MetaData16$genoID<-as.factor(MetaData16$genoID)
MetaData16$genetID<-as.factor(MetaData16$genetID)
MetaData16$Tank<-as.factor(MetaData16$Tank)
MetaData16$type<-as.factor(MetaData16$type)
MetaData16$symbiont<-as.factor(MetaData16$symbiont)

sam0 <- MetaData16
sam1 <- as.matrix(sam0[, -1])
rownames(sam1) <- sam0$sample_name
sam <- sample_data(data.frame(sam1))

#load in OUT:raw_abundanceTable_100.shared
OTU3k<-read.table("16S-pipeline_outputs/Results/raw/details/raw_abundanceTable_100.shared", sep='', header=T)

OTU3k <- as.matrix(OTU3k[, -(1)]) # remove first col "label"
# replace the "Group" col with actual sample IDs.
indexes<-MetaData16[,1:2]
otu3k2<-merge(indexes,OTU3k, by="Group") #add sample names
otu3k2<-otu3k2[,-(3)]# remove first col "Group" and num Otu
otu3k2<-otu3k2[,-(1)]
otu3k2 <- as.matrix(otu3k2) #make matrix
otu3k2.df<-as.data.frame(otu3k2) #make df copy to make samp names as row names in matrix
rownames(otu3k2) <- otu3k2.df$sample_name
otu3k2 <- as.matrix(otu3k2[, -(1)]) # remove col sample_name

## something is preventing phyloseq from taking otu3k2 as the otu table. but if you save and then reupload it works. 
#write.csv(otu3k2,"otu3k2check.csv")
testOtu<-read.csv("otu3k2check.csv")
testOtu2 <- as.matrix(testOtu[, -(1)]) 
rownames(testOtu2)<- otu3k2.df$sample_name
otu <- otu_table(testOtu2, taxa_are_rows = FALSE)

#tax table annotations_100_taxonomy.csv (edited to be in proper format with proper col names in excel (remove ";"))
TAX<- read.csv("16S-pipeline_outputs/Results/main/details/annotations_100.taxonomy_format.csv", colClasses = "character") #copied from 16S and put in main folder (trouble accessing)

tax1 <- as.matrix(TAX[, -1], dimnames = list(TAX$OTU, colnames(TAX[-1])))
rownames(tax1) <- TAX$OTU
tax <- tax_table(tax1)

# Read the data into phyloseq
Bac.seq = phyloseq(otu,sam, tax) 
Bac.seq
Bac.seq.df <- sample_data(Bac.seq)

#load your tree: "FastTree_100.nwk"
treefile<- read.tree("16S-pipeline_outputs/Results/postprocessing/unifrac/FastTree_100.nwk")
phy_tree(Bac.seq) <- treefile
Bac.seq

save(Bac.seq, file = "Data/Bac.seq_phyloseq.RData")
```

#Start analysis here: load in RData if needed
```{r}
#load("Data/Bac.seq_phyloseq.RData")
```


#QC data
```{r}
# the df is Bac.seq
#Vew data
rank_names(Bac.seq) #ranks are DIV and clade
ntaxa(Bac.seq)  #num taxa
nsamples(Bac.seq)   #num samples
sample_names(Bac.seq) #samp names
sample_variables(Bac.seq) # metadata cats
# create df of sample data to view 
sample.data <- as(sample_data(Bac.seq), "data.frame") #create sample data frame to view
sample.data$LibrarySize <- sample_sums(Bac.seq)
sample.data <- sample.data[
  order(sample.data$LibrarySize),]
sample.data$Index <- seq(nrow(sample.data))  # check 3k for all 
ggplot(data = sample.data, aes(x=Index, y=LibrarySize, color = sample_name)) +
  geom_point()

sample_sum_df <- data.frame(sum = sample_sums(Bac.seq)) #look a seq depth. 

#NTC
NTCs<-subset_samples(Bac.seq, type=="NTC")
TopNOTUsNTC = names(sort(taxa_sums(NTCs), TRUE)[1:200])
bac50 = prune_taxa(TopNOTUsNTC, NTCs)
bacBarPlot<-plot_bar(bac50,  fill="Phylum");bacBarPlot

#Mock ADD THIS IN LATER
# Mock<-subset_samples(Bac.seq, sample.type=="Mock")
# TopNOTUsMock = names(sort(taxa_sums(NTCs), TRUE)[1:200])
# bac50 = prune_taxa(TopNOTUsMock, Mock)
# bacBarPlot<-plot_bar(bac50,  fill="Phylum");bacBarPlot
# 
# richness(Mock)
# plot_richness(Mock, measures = c("Observed","Shannon"))
# evenness(Mock)

```
Making rare and non-rare dataframes (remove low samples)
```{r}

#make a non-rarefied df:
#Remove all samples with <400 reads and create sample data. if we cut under 1000, we lose most of Mote
#These data also useful for beta diversity analyses (non-rarefied)
Bac_nonrare <- prune_samples(sample_sums(Bac.seq)>=400, Bac.seq) # remove samps <2000, save as new obj
Bac_nonrare_sd <- as(sample_data(Bac_nonrare), "data.frame")

#Make a rarefied phyloseq object for alpha diversity analyses
Bac_rare <- rarefy_even_depth(Bac.seq, sample.size = 400, rngseed = 711)   #C14 sample removedbc less than 2000 
sample_sums(Bac_rare) #Double check that the sub-sampling worked, this should report 2000 for each sample
Bac_rare_sd <- as(sample_data(Bac_rare), "data.frame")
```

Turn rarified df in to relative abundance:  
```{r}
Bac_rare_RelA  = transform_sample_counts(Bac_rare, function(x) x / sum(x) )  #save as RELA DIVs
Bac_rare_RelA_sd <- as(sample_data(Bac_rare_RelA), "data.frame") #look at samp data
Bac_rare_RelA = prune_taxa(taxa_sums(Bac_rare_RelA) > 0, Bac_rare_RelA) #this removes any OTU with 0s
   
#prune the non-rarified data
Bac_nonrare = prune_taxa(taxa_sums(Bac_nonrare) > 0, Bac_nonrare) #this removes any OTU with 0s  
  
```

#Alpha diversity use non-relA df: Bac_nonrare
```{r}
Bac_nonrare_sd <- as(sample_data(Bac_nonrare), "data.frame") #sample df

# Do in microbiome package (USING THIS) ####
erich.tab <-microbiome::alpha(Bac_nonrare, index = "all") # pull out: observed, diversity_shannon, evenness_pielou
erich.tab2<-erich.tab[-c(2:4,6:8,10:22)] #keep the cols you need

 erich.tab2$sample_name <- as.factor(Bac_nonrare_sd$sample_name)       #add back in meta
  erich.tab2$nursery <- as.factor(Bac_nonrare_sd$nursery)
  erich.tab2$parentID <- as.factor(Bac_nonrare_sd$parentID)
  erich.tab2$symbiont <- as.factor(Bac_nonrare_sd$symbiont)
  
#ANOVAS
#shannon ####
  hist(erich.tab2$diversity_shannon)

  set.seed(30)
  aov.shannon.species = aov(diversity_shannon ~ symbiont, data=erich.tab2)
    summary(aov.shannon.species) # significant
      TukeyHSD(aov.shannon.species)

  set.seed(30)
    aov.shannon.species = aov(diversity_shannon ~ parentID, data=erich.tab2) 
      summary(aov.shannon.species) # not diff by parentID

    #look at residuals
    car::qqPlot(residuals(aov.shannon.species))
    car::leveneTest(residuals(aov.shannon.species)~symbiont, data=erich.tab2) #passes
    
  shannon_mean <- ddply(erich.tab2, c("symbiont"), summarise, #summarize cell counts
                 N    = length(diversity_shannon[!is.na(diversity_shannon)]), #calculate the length of the data frame, excluding NA’s
                 mean = mean(diversity_shannon, na.rm=TRUE), #calculate mean of response variable, removing NA's
                 sd   = sd(diversity_shannon, na.rm=TRUE), #calculate standard deviation
                 se   = sd / sqrt(N), #calculate standard error
                 max = max(diversity_shannon, na.rm=TRUE) #calculate max, could also calculate min () if desired
);shannon_mean

shannon_plot<-ggplot(data=shannon_mean, aes(x=symbiont, y=mean, color=symbiont)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=symbiont), size=4, show.legend = F)+
  xlab("") + #Label the X Axis
    ylab("") + #Label the X Axis
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  # theme(panel.grid.major = element_blank(), 
  #       panel.grid.minor = element_blank(),
  #       panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
   ylab(expression(paste("Shannon Diversity"))) +
   ggtitle("Shannon")+
    theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+
  theme(plot.title = element_text(size=20, face = "italic"));shannon_plot

#observed   ####
   hist(erich.tab2$observed) 
  ggqqplot(erich.tab2$observed)
  
  erich.tab2$observed_log<-log10(erich.tab2$observed)
  hist(erich.tab2$observed_log) 
  
  set.seed(30)
    aov.observed.species = aov(observed_log ~ symbiont, data=erich.tab2)
      summary(aov.observed.species) # almost 0.051
  
  observed_mean <- ddply(erich.tab2, c("symbiont"), summarise, #summarize cell counts
                 N    = length(observed[!is.na(observed)]), #calculate the length of the data frame, excluding NA’s
                 mean = mean(observed, na.rm=TRUE), #calculate mean of response variable, removing NA's
                 sd   = sd(observed, na.rm=TRUE), #calculate standard deviation
                 se   = sd / sqrt(N), #calculate standard error
                 max = max(observed, na.rm=TRUE) #calculate max, could also calculate min () if desired
);observed_mean

observed_plot<-ggplot(data=observed_mean, aes(x=symbiont, y=mean, color=symbiont)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=symbiont), size=4, show.legend = F)+
  xlab("") + #Label the X Axis
    ylab("") + #Label the X Axis
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  # theme(panel.grid.major = element_blank(), 
  #       panel.grid.minor = element_blank(),
  #       panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
   ylab(expression(paste("observed Diversity"))) +
   ggtitle("observed")+
    theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+
  theme(plot.title = element_text(size=20, face = "italic"));observed_plot
 
 #evenness ####

 hist(erich.tab2$evenness_pielou)
  ggqqplot(erich.tab2$evenness_pielou)
   hist(erich.tab2$evenness_pielou_log10)#not great 

  set.seed(30)
    aov.evenness_pielou.species = aov(evenness_pielou ~ symbiont*parentID, data=erich.tab2)
      summary(aov.evenness_pielou.species) # Tag ID and location significant
  
    car::qqPlot(residuals(aov.evenness_pielou.species))
    car::leveneTest(residuals(aov.evenness_pielou.species)~symbiont, data=erich.tab2) #fails

    #transformed data also fails, stick with original data (sqrt fails, log fails, in fails, arcsqrt fails)
    
  aov.evenness_pielou.species = aov(evenness_pielou ~ date, data=erich.tab2)
  summary(aov.evenness_pielou.species) #not sig
  
  aov.evenness_pielou.species = aov(evenness_pielou ~ location, data=erich.tab2)
  summary(aov.evenness_pielou.species) # sig
  TukeyHSD(aov.evenness_pielou.species)#

  aov.evenness_pielou.species = aov(evenness_pielou ~ tag.id, data=erich.tab2)
  summary(aov.evenness_pielou.species) # p < 0.001
  TukeyHSD(aov.evenness_pielou.species)# 
  
  aov.evenness_pielou.species = aov(evenness_pielou ~ Prop.D, data=erich.tab2)
  summary(aov.evenness_pielou.species) # p =.43
  
  evenness_pielou_mean <- ddply(erich.tab2, c("symbiont"), summarise, #summarize cell counts
                 N    = length(evenness_pielou[!is.na(evenness_pielou)]), #calculate the length of the data frame, excluding NA’s
                 mean = mean(evenness_pielou, na.rm=TRUE), #calculate mean of response variable, removing NA's
                 sd   = sd(evenness_pielou, na.rm=TRUE), #calculate standard deviation
                 se   = sd / sqrt(N), #calculate standard error
                 max = max(evenness_pielou, na.rm=TRUE) #calculate max, could also calculate min () if desired
);evenness_pielou_mean

evenness_pielou_plot<-ggplot(data=evenness_pielou_mean, aes(x=symbiont, y=mean, color=symbiont)) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se),
                width=0.1, show.legend = F)+
  geom_point(aes(color=symbiont), size=4, show.legend = F)+
  xlab("") + #Label the X Axis
    ylab("") + #Label the X Axis
  theme_bw() + #Set the background color
  theme(axis.line = element_line(color = 'black'), #Set the axes color
        axis.title=element_text(size=14,face="bold"), #Set axis format
        panel.border = element_blank(), #Set the border
        panel.grid.major = element_blank(), #Set the major gridlines
        panel.grid.minor = element_blank(), #Set the minor gridlines
        text = element_text(size=18),  # set element text
        plot.background =element_blank(), #Set the plot background
        legend.key = element_blank()) + #Set plot legend key
  # theme(panel.grid.major = element_blank(), 
  #       panel.grid.minor = element_blank(),
  #       panel.background = element_rect(colour = "black", size=1))+
        theme(aspect.ratio=1)+
   ylab(expression(paste("evenness_pielou Diversity"))) +
   ggtitle("evenness_pielou")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+
  theme(plot.title = element_text(size=20, face = "italic"));evenness_pielou_plot 
  
#plot together  

grid.arrange( observed_plot, evenness_pielou_plot,shannon_plot, nrow = 1)

  
#regular box  
#evenness
  aov.evenness.species = aov(evenness_pielou ~ symbiont, data=erich.tab2)
  summary(aov.evenness.species) # sig diff between species
  TukeyHSD(aov.evenness.species)  

  #Plot
  boxplot(diversity_shannon ~ symbiont, data=erich.tab2, ylab="Shannon's diversity") 
  boxplot(observed ~ location, data=erich.tab2, ylab="Observed") 
  boxplot(evenness_pielou ~ location, data=erich.tab2, ylab="Evenness_pielou") 

#violin plots
  # create a list of pairwise comaprisons
al.species <- levels(erich.tab2$symbiont) # get the variables

# make a pairwise list that we want to compare.
  al.species.pairs <- combn(seq_along(al.species), 2, simplify = FALSE, FUN = function(i)al.species[i])
  print(al.species.pairs)

#observed
p1 <- ggviolin(erich.tab2, x = "symbiont", y = "observed",
   add = "boxplot", fill = "symbiont") 
  print(p1)

    p1 <- ggviolin(erich.tab2, x = "tag.id", y = "observed",
   add = "boxplot", fill = "symbiont") 
  print(p1)

  #Shannon
p1 <- ggviolin(erich.tab2, x = "symbiont", y = "diversity_shannon",
   add = "boxplot", fill = "symbiont") 
  print(p1)
  
p1 <- p1 + stat_compare_means(comparisons = "tag.id") #non-parametric test (Wilcoxon test)
  print(p1)
  
p1 <- ggviolin(erich.tab2, x = "tag.id", y = "diversity_shannon",
   add = "boxplot", fill = "symbiont") 
  print(p1)
  
#Evenness
p1 <- ggviolin(erich.tab2, x = "symbiont", y = "evenness_pielou",
   add = "boxplot", fill = "symbiont") 
  print(p1)
  
  
p1 <- p1 + stat_compare_means(comparisons = "symbiont") #non-parametric test (Wilcoxon test)
  print(p1)



```

#Stats: beta diversity
```{r}
#make unifrac matrix
Bac.wu <- phyloseq::distance(Bac_rare_RelA, method = "wunifrac")
samp.df <- as(sample_data(Bac_rare_RelA), "data.frame") #sample df
samp.df$sample_name<-as.factor(samp.df$sample_name)
samp.df$nursery<-as.factor(samp.df$nursery)
samp.df$parentID<-as.factor(samp.df$parentID)
samp.df$symbiont<-as.factor(samp.df$symbiont)


#adonis

#between symbionts and parents 
set.seed(30)
adonis2(Bac.wu ~ symbiont, data = samp.df) #sig

set.seed(30)
adonis2(Bac.wu ~ parentID, data = samp.df) #not sig

# Homogeneity of dispersion test  
    set.seed(30) # symbiont
    permutest(betadisper(Bac.wu, samp.df$symbiont, type = "centroid"))  # p =0.001
  ball<-  betadisper(Bac.wu, samp.df$symbiont, type = "centroid")
  TukeyHSD(ball,  ordered = FALSE,conf.level = 0.95)
  
  
```
NMDS

```{r}
#nmds

unifrac.b <- ordinate(Bac_rare_RelA, method="NMDS", "unifrac",weighted=TRUE, set.seed(30)) 
stressplot(unifrac.b)
scores.b<-as.data.frame(cbind(vegan::scores(unifrac.b, display="sites")))  
     unifrac.sd <- as(sample_data(Bac_rare_RelA), "data.frame") #sample df

scores.b$sample_name <- as.factor(unifrac.sd$sample_name)
scores.b$symbiont <- as.factor(unifrac.sd$symbiont)
scores.b$parentID <- as.factor(unifrac.sd$parentID)

###### plots

ggplot(scores.b, aes(x = NMDS1, y = NMDS2, fill=symbiont)) + 
  geom_point(aes(color=symbiont),size = 4) +
    scale_color_manual(values=c("orange", "dodgerblue3"))+
stat_ellipse(geom="polygon", alpha=1/6, aes(color=symbiont)) +
  ggtitle("NMDS 16s") +
  theme_classic()

ggplot(scores.b, aes(x = NMDS1, y = NMDS2, fill=symbiont)) + 
  geom_point(aes(color=symbiont),size = 2) +
    scale_color_manual(values=c("orange", "dodgerblue3"))+
  ggtitle("NMDS 16s") +
  theme_classic()+
  facet_wrap(~parentID)
```

# who is there: bar plots Phylum
```{r}
Bac.phy <- Bac_rare_RelA %>% 
  tax_glom(taxrank = "Phylum", NArm = FALSE) %>% 
  transform_sample_counts(function(x) {x/sum(x)} ) 
Bac.phy.melt <- psmelt(Bac.phy)

nb.cols <- 48
mycolors <- colorRampPalette(brewer.pal(11, "RdYlBu"))(nb.cols)

pal = qualpal(48, colorspace=list(h=c(13,350), s=c(0.3,1), l=c(0.2,0.8)))

apal.p <- ggplot(Bac.phy.melt, aes(x = parentID, y = Abundance, fill = Phylum)) +
  geom_bar(stat = "identity") +
  #scale_fill_manual(values=pal) +
          scale_color_manual(values=pal$hex)+
  ylab("Relative Abundance") +
  xlab("location") +
  #theme(legend.position = "none")+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
apal.p +  facet_wrap(~ symbiont, nrow=2)



#AH to get percent taxa by grouping
dat <- Bac_rare_RelA %>% tax_glom(taxrank = "Phylum") %>% psmelt()
head(dat)

df<-dat %>% 
  group_by(symbiont, Phylum) %>% 
  summarise(TotalAbundance=sum(Abundance))%>% #calculate total abundance of each phylum
  group_by(symbiont) %>%
  mutate(PercRelAbundance=TotalAbundance/sum(TotalAbundance)*100) %>%#calculate relative abundance
  dplyr::select(!TotalAbundance) 

df$Phylum <- gsub('[^[:alnum:] ]','',df$Phylum)

#change to a matrix with phylum in rows and lifestage in columns
df<-df%>%
  spread(key=Phylum, value=PercRelAbundance)
#replace na with 0, since 0 abundance was calculated as na in step above b/c dividing by 0
df[is.na(df)] <- 0
rownames(df)<-df$symbiont
location_list<-df$symbiont
dim(df)
df<-as.matrix(df)
df<-df[,-1]
taxa_list<-colnames(df)
df_mat <- matrix(as.numeric(df),    # Convert to numeric matrix
                  ncol = ncol(df))                                   # Print numeric matrix
colnames(df_mat)<-taxa_list
rownames(df_mat)<-location_list
df_mat<-t(df_mat)
### above works

df_mat_long<-melt(df_mat, id.vars=c("")) #long format
df_mat_long<-as.data.frame(df_mat_long)

p<-ggplot(df_mat_long, aes(x = Var2, y = value, fill=value)) + 
  geom_bar(stat="identity")+
  ggtitle("Top Phyla") +
  theme_classic()

 p+ facet_wrap(~Var1)

 

```
Try heatmap phylum
```{r}
 
library(ComplexHeatmap)
library(dendsort)
library(colorRamp2)
row_dend = dendsort(hclust(dist(df_mat_long)))
col_dend = dendsort(hclust(dist(t(df_mat_long))))
col_fun = colorRamp2(c(0, 50, 100), c("white", "darkgray", "navy"))
pdf(file = "16S-Phylum-Heatmap.pdf", height = 8, width = 18)
ComplexHeatmap::Heatmap(df_mat_long, name = "Rel. Abun. (%)", row_title = "", column_title = "16S Relative Abundance (Phylum)", 
        col = col_fun, 
        row_names_side = "left", 
        row_dend_side = "right",
        width = unit(4, "in"), 
        height = unit(4, "in"), 
        column_dend_height = unit(.5, "in"),
        row_dend_width = unit(.5, "in"),
        column_dend_reorder = FALSE,
        row_dend_reorder = TRUE,
        row_gap = unit(2.5, "mm"),
        clustering_distance_rows = "euclidean",
        clustering_method_rows = "complete",
        border = TRUE, 
        column_names_gp =  gpar(fontsize = 12, border=FALSE),
        column_names_rot = 35,
        cluster_rows = row_dend, 
        #cluster_columns = col_dend,
        row_names_gp = gpar(fontsize = 12, alpha = 0.75, border = FALSE))
dev.off()

plot<-plot_bar(Bac, fill = "Phylum")+theme(legend.position="bottom")
```
Family 
```{r}

Bac.fam <- Bac_rare_RelA %>% 
  tax_glom(taxrank = "Family", NArm = FALSE) %>% 
  transform_sample_counts(function(x) {x/sum(x)} ) 
Bac.fam.melt <- psmelt(Bac.fam)

Bac.fam.melt$Code<-paste(Bac.fam.melt$symbiont, Bac.fam.melt$parentID)
Bac.fam.melt$Code <- factor(Bac.fam.melt$Code)



nb.cols <- 48
mycolors <- colorRampPalette(brewer.pal(11, "RdYlBu"))(nb.cols)

pal = qualpal(48, colorspace=list(h=c(13,350), s=c(0.3,1), l=c(0.2,0.8)))



mcap.p <- ggplot(Bac.fam.melt, aes(x = sample_name, y = Abundance, fill = Family)) +
  geom_bar(stat = "identity") +
  #scale_fill_manual(values=pal) +
          scale_color_manual(values=pal$hex)+
  ylab("Relative Abundance") +
  xlab("location") +
  theme(legend.position = "none")+
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
mcap.p +  facet_wrap(~ symbiont, nrow=2)

```

