---
title: "R Notebook"
output: html_notebook
editor_options: 
  chunk_output_type: console
---
## Note, remove samples tube 129 and 134, were resampled in other tubes

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
#16S Genotype swap data first pass

```{r}  
knitr::opts_chunk$set(warning=FALSE, message=FALSE)


library(plyr) 
library(dplyr)
library(ggplot2)
library(ggpubr)
library(lmerTest)
library(car)
library(emmeans)
library(gridExtra)
library(multcomp)
library(reshape)
library(factoextra)
library(reshape2)
library(vegan) 
library(pairwiseAdonis)
library("scales")
packageVersion("scales")
library(RColorBrewer)
library(colorRamps)
library(devtools)
library(phyloseq)
library(readr)
library(vegan)
library(ape)
library(geosphere)
library(ade4)
library(microbiome)  
library(knitr)

```

#Load in data and get into phyloseq (save RData file)
```{r}       
#read in sample data and merge metadata
MetaData16<-read.csv("genoswap_metadata.csv") #same for ITS2
MetaData16$species<-as.factor(as.character(MetaData16$species))
#MetaData16$tube.id<-as.numeric(as.character(MetaData16$tube.id))
MetaData16$type<-as.factor(as.character(MetaData16$type))
MetaData16$duplicates<-as.factor(as.character(MetaData16$duplicates))
MetaData16<-MetaData16[c(1:5)] #keep cols you need
MetaData16<-MetaData16[!(MetaData16$species=="Apal"),] #remove the APAL 


#read in metadata for samp collections
genSamp<-read.csv("data/geneticSamples.csv") #load in Kelsey/Rish nursery data
# notes: swap.genotype is the name to use. the "ziploc genet" name is either bag UM collected in or the changed name a specific nursery calls the genotype. Swap.geno is the name to use here. Rich confirmed all cols to the right of 'notes' unneeded
genSamp2<-genSamp[c(1,4,5)] #keep cols you need
colnames(genSamp2)[3] ="tube.id" #change col name to sample tube ID
#write.csv(genSamp2,"genSamp2.csv") The metadata has an issue where the genet name is entered slighty different by nursery so, here, we are going with the genoHome genet names. Download the csv and spot check and edit the genet names in THIS file. Also remove tube.ID with NAs
genSamp3<-read.csv("genSamp3.csv") #upload cleaned and QC-ed geno names
genSamp3$genet.ID<-gsub("-","",as.character(genSamp3$genet.ID)) #remove all "-", " ", "'"
genSamp3$genet.ID<-gsub(" ","",as.character(genSamp3$genet.ID)) #remove all "-", " ", "'"
genSamp3$genet.ID<-gsub("'","",as.character(genSamp3$genet.ID)) #remove all "-", " ", "'"



#upload meta for Fresh samples to merge add to genSamp3
genSamp.Fresh<-read.csv("Fresh.meta.away.csv") #load in Kelsey/Rish nursery data
  genSamp.Fresh$genet.ID<-gsub("-","",as.character(genSamp.Fresh$genet.ID)) #remove all "-", " ", "'"
  genSamp.Fresh$genet.ID<-gsub(" ","",as.character(genSamp.Fresh$genet.ID)) #remove all "-", " ", "'"
  genSamp.Fresh$genet.ID<-gsub("'","",as.character(genSamp.Fresh$genet.ID)) #remove all "-", " ", "'"

#read in home location for each genet (removed lat/long due to symblols)
Genotype.Home<-read.csv("Genotype.Home.csv") #load in Kelsey/Rish nursery data
Genotype.Home<-Genotype.Home[c(1:2)] #keep cols you need
colnames(Genotype.Home)[1] ="genet.ID" #change col name to sample tube ID
colnames(Genotype.Home)[2] ="Home.nursery"


  #### There are some with no tube ID, means no samples. remove rows with no tube id. HAVE NOT DONE
#write.csv(Genotype.Home,"Genotype.Home.csv")
  Genotype.Home$genet.ID<-gsub("-","",as.character(Genotype.Home$genet.ID)) #remove all "-", " ", "'"
  Genotype.Home$genet.ID<-gsub(" ","",as.character(Genotype.Home$genet.ID)) #remove all "-", " ", "'"
  Genotype.Home$genet.ID<-gsub("'","",as.character(Genotype.Home$genet.ID)) #remove all "-", " ", "'"
Genotype.Home<-na.omit(Genotype.Home)


genSamp4<-merge(genSamp3,Genotype.Home, by="genet.ID", all.x=T)

genSamp5<-rbind(genSamp4, genSamp.Fresh)
genSamp5$Home.nursery[genSamp5$genet.ID == "ML1"] <- "Mote" ### figure out how to code this better
genSamp5$Home.nursery[genSamp5$genet.ID == "ML13"] <- "Mote"
genSamp5$Home.nursery[genSamp5$genet.ID == "ML29"] <- "Mote"
genSamp5$Home.nursery[genSamp5$genet.ID == "ML3"] <- "Mote"
genSamp5$Home.nursery[genSamp5$genet.ID == "ML31"] <- "Mote"
genSamp5$Home.nursery[genSamp5$genet.ID == "ML34"] <- "Mote"
genSamp5$Home.nursery[genSamp5$genet.ID == "ML41"] <- "Mote"
genSamp5$Home.nursery[genSamp5$genet.ID == "ML5"] <- "Mote"
genSamp5$Home.nursery[genSamp5$genet.ID == "ML50"] <- "Mote"
genSamp5$Home.nursery[genSamp5$genet.ID == "ML55"] <- "Mote"
genSamp5$Home.nursery[genSamp5$genet.ID == "ML57"] <- "Mote"
genSamp5$Home.nursery[genSamp5$genet.ID == "ML6"] <- "Mote"
genSamp5$Home.nursery[genSamp5$genet.ID == "ML62"] <- "Mote"
genSamp5$Home.nursery[genSamp5$genet.ID == "ML7"] <- "Mote"
genSamp5$Home.nursery[genSamp5$genet.ID == "ML84"] <- "Mote"
genSamp5$Home.nursery[genSamp5$genet.ID == "ML89"] <- "Mote"
genSamp5$Home.nursery[genSamp5$genet.ID == "ML90"] <- "Mote"

#Combine all metadata
meta<-merge(MetaData16,genSamp5, by="tube.id", all.x=T) #merge all
#write.csv(meta, "meta.csv")

meta$frag.id<-paste(meta$genet.ID, meta$Home.nursery, meta$Away.nursery.x)
  meta$tube.id<-as.factor(as.character(meta$tube.id))
  meta$type<-as.factor(as.character(meta$type))
  meta$duplicates<-as.factor(as.character(meta$duplicates))
  meta$species<-as.factor(as.character(meta$species))
  meta$genet.ID<-as.factor(as.character(meta$genet.ID))
  meta$Away.nursery<-as.factor(as.character(meta$Away.nursery))
  meta$Home.nursery<-as.factor(as.character(meta$Home.nursery))
#write.csv(meta, "meta.csv") #check 
  
# create new column called frag.id that is a concat of genetID-Home-Away
#write.csv(MetaData16, "Metadata16index.csv") #edited for index duplicates and loaded above

sam0 <- meta
sam1 <- as.matrix(sam0[, -1])
rownames(sam1) <- sam0$sample_name
sam <- sample_data(data.frame(sam1))
write.csv(sam, "sam.csv")
#load in OTU: raw_abundanceTable_100.shared
OTU3k<-read.table("data/16Stake4/16S-pipeline_outputs/Results/raw/details/raw_abundanceTable_100.shared.txt", sep='', header=T)

test<-OTU3k
test$id<-sub("_*", "", test$Group)
#write.csv(test, "test.csv") write and use excel to create a index for group. figure out how to do this for real. 

#otu
otu3k1 <- as.matrix(OTU3k[, -(1:3)]) # remove first col "label"
# replace the "Group" col with actual sample IDs.
# make a file with the "group" ids and the sample ids for metadata
indexes<-read.csv("data/indexes_take4.csv")   #upload the sample ids
otu3k2<-merge(indexes,OTU3k, by="Group") #add sample names
otu3k2 <- as.matrix(otu3k2[, -c(1,3:4)]) # remove first col "Group", and label and num OTU
otu3k2.df<-as.data.frame(otu3k2) #make df copy to make samp names as row names in matrix

rownames(otu3k2) <- otu3k2.df$sample_name
otu3k2 <- as.matrix(otu3k2[, -(1)]) # remove first col samplename

## something is preventing phyloseq from taking otu3k2 as the otu table. but if you save and then reupload it works. 
#write.csv(otu3k2,"otu3k2check.csv")
testOtu<-read.csv("otu3k2check.csv")
testOtu2 <- as.matrix(testOtu[, -(1)]) 
rownames(testOtu2)<- otu3k2.df$sample_name
otu <- otu_table(testOtu2, taxa_are_rows = FALSE)

#tax table annotations_100_taxonomy.csv (edited to be in proper format with proper col names in excel (remove ";"))
TAX<- read.csv("data/raw_consensusClassification_100.taxonomy.csv", colClasses = "character") 
tax1 <- as.matrix(TAX[, -1], dimnames = list(TAX$OTU, colnames(TAX[-1])))
rownames(tax1) <- TAX$OTU
tax <- tax_table(tax1)

# Read the data into phyloseq
Bac.seq = phyloseq(otu, tax,sam) #THIS WORKS
Bac.seq
Bac.seq.df <- sample_data(Bac.seq)

#load your .tre otu_repr_100.tre
#treefile<- read.tree("MCL19Summer_16S-pipeline_outputs USE THIS ONE/Results/postprocessing/unifrac/otu_repr_100.tre")
#phy_tree(Bac.seq) <- treefile
#Bac.seq

save(Bac.seq, file = "data/Bac.seq_phyloseq.RData")
```
7/20/23 there is unexpectedly no .tre file in the 16S outputs. I am re-running the CMAIKI cluster with same parameters to see if there was an issue. 