---
title: "R Notebook"
output: html_notebook
editor_options: 
  chunk_output_type: console
---


```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
#16S Genotype swap data first pass

```{r}  
knitr::opts_chunk$set(warning=FALSE, message=FALSE)


library(plyr) 
library(dplyr)
library(ggplot2)
library(ggpubr)
library(lmerTest)
library(car)
library(emmeans)
library(gridExtra)
library(multcomp)
library(reshape)
library(factoextra)
library(reshape2)
library(vegan) 
library(pairwiseAdonis)
library("scales")
packageVersion("scales")
library(RColorBrewer)
library(colorRamps)
library(devtools)
library(phyloseq)
library(readr)
library(vegan)
library(ape)
library(geosphere)
library(ade4)
library(microbiome)  
library(knitr)

```

#Load in data and get into phyloseq (save RData file)
```{r}       
#read in sample data and merge metadata
MetaData16<-read.csv("genoswap_metadata.csv") #same for ITS2
MetaData16$species<-as.factor(as.character(MetaData16$species))
MetaData16$tube.id<-as.factor(as.character(MetaData16$tube.id))
MetaData16$type<-as.factor(as.character(MetaData16$type))
MetaData16$duplicates<-as.factor(as.character(MetaData16$duplicates))
MetaData16<-MetaData16[c(1:5)] #keep cols you need

#read in metadata for samp collections
genSamp<-read.csv("data/geneticSamples.csv") #load in Kelsey/Rish nursery data
# notes: swap.genotype is the name to use. the "ziploc genet" name is either bag UM collected in or the changed name a specific nursery calls the genotype. Swap.geno is the name to use here. Rich confirmed all cols to the right of 'notes' unneeded
genSamp2<-genSamp[c(2:8)] #keep cols you need
colnames(genSamp2)[5] ="tube.id" #change col name to sample tube ID
colnames(genSamp2)[1] ="Away.nursery"
colnames(genSamp2)[3] ="Away.nursery.ID"
colnames(genSamp2)[4] ="genet.ID"


#read in home location for each genet (removed lat/long due to symblols)
Genotype.Home<-read.csv("data/genoHome.csv") #load in Kelsey/Rish nursery data
Genotype.Home<-Genotype.Home[c(1:2, 4:8)] #keep cols you need
colnames(Genotype.Home)[1] ="genet.ID" #change col name to sample tube ID
colnames(Genotype.Home)[2] ="Home.nursery"
colnames(Genotype.Home)[3] ="Other.name"

#Combine all metadata
metadata1<-merge(genSamp2,Genotype.Home, by="genet.ID", all.x=TRUE) #merge
meta<-merge(MetaData16,metadata1, by="tube.id", all.x=T) #merge all

meta$frag.id<-paste(meta$genet.ID, meta$Home.nursery, meta$Away.nursery.x)
  meta$tube.id<-as.factor(as.character(meta$tube.id))
  meta$type<-as.factor(as.character(meta$type))
  meta$duplicates<-as.factor(as.character(meta$duplicates))
  meta$species<-as.factor(as.character(meta$species))
  meta$genet.ID<-as.factor(as.character(meta$genet.ID))
  meta$Away.nursery<-as.factor(as.character(meta$Away.nursery))
  meta$Home.nursery<-as.factor(as.character(meta$Home.nursery))

# create new column called frag.id that is a concat of genetID-Home-Away



#write.csv(MetaData16, "Metadata16index.csv") edited for index duplicates and loaded above

sam0 <- MetaData16
sam1 <- as.matrix(sam0[, -1])
rownames(sam1) <- sam0$sample_name
sam <- sample_data(data.frame(sam1))

#load in OTU: raw_abundanceTable_100.shared
OTU3k<-read.table("16S-pipeline_outputs 2/Results/raw/details/raw_abundanceTable_100.shared.txt", sep='', header=T)

test<-OTU3k
test$id<-sub("_*", "", test$Group)
#write.csv(test, "test.csv") write and use excel to create a index for group. figure out how to do this for real. 

#otu
otu3k1 <- as.matrix(OTU3k[, -(1:3)]) # remove first col "label"
# replace the "Group" col with actual sample IDs.
# make a file with the "group" ids and the sample ids for metadata


indexes<-read.csv("indexOtu.csv")   #upload the sample ids
otu3k2<-merge(indexes,OTU3k, by="Group") #add sample names
otu3k2 <- as.matrix(otu3k2[, -c(1,3:4)]) # remove first col "Group", and label and num OTU
otu3k2.df<-as.data.frame(otu3k2) #make df copy to make samp names as row names in matrix

rownames(otu3k2) <- otu3k2.df$sample_name
otu3k2 <- as.matrix(otu3k2[, -(1)]) # remove first col samplename

## something is preventing phyloseq from taking otu3k2 as the otu table. but if you save and then reupload it works. 
#write.csv(otu3k2,"otu3k2check.csv")
testOtu<-read.csv("otu3k2check.csv")
testOtu2 <- as.matrix(testOtu[, -(1)]) 
rownames(testOtu2)<- otu3k2.df$sample_name
otu <- otu_table(testOtu2, taxa_are_rows = FALSE)

#tax table annotations_100_taxonomy.csv (edited to be in proper format with proper col names in excel (remove ";"))
TAX<- read.csv("raw_consensusClassification_100.taxonomy.csv", colClasses = "character") 
tax1 <- as.matrix(TAX[, -1], dimnames = list(TAX$OTU, colnames(TAX[-1])))
rownames(tax1) <- TAX$OTU
tax <- tax_table(tax1)

# Read the data into phyloseq
Bac.seq = phyloseq(otu, tax,sam) #THIS WORKS
Bac.seq
Bac.seq.df <- sample_data(Bac.seq)

#load your .tre otu_repr_100.tre
#treefile<- read.tree("MCL19Summer_16S-pipeline_outputs USE THIS ONE/Results/postprocessing/unifrac/otu_repr_100.tre")
#phy_tree(Bac.seq) <- treefile
#Bac.seq

save(Bac.seq, file = "data/Bac.seq_phyloseq.RData")
```
7/20/23 there is unexpectedly no .tre file in the 16S outputs. I am re-running the CMAIKI cluster with same parameters to see if there was an issue. 